### PR TITLE
Fix bug that wipes out existing form values with invalid CSV uploads

### LINKED ISSUE #4323: Registration form becomes non-functional after CSV upload submission failure
**Describe the bug**
If the user encounters an error on the registration from using the CSV upload, the form reports an error and then becomes non-functional.

I've made a screencast illustrating the problem. These are the steps I followed:

1. Choose an APO and enter a Project tag on the form
2. Choose CSV upload
3. Upload a CSV that triggers a validation error
4. Received validation error (displayed on the form)
5. Noted also that the form cleared my APO and Project tag selections
6. Was unable to upload a fixed CSV and resubmit the form

https://github.com/sul-dlss/argo/assets/8975583/4ad31188-db90-41c2-ad49-1b2e34fb9e64

**User Impact**
The user is left wondering why the form does not work even after they fix their CSV. There's no indication that they will have to reload the page and start over.

**To Reproduce**

Use this CSV: 
[registration-break.csv](https://github.com/sul-dlss/argo/files/14516392/registration-break.csv)

Then follow the steps in the screencast described above.

**Expected behavior**

Ideal case:

After receiving an error message, the form should:

1. Retain the previously entered data on the top half of the form (APO choice, tags, etc.)
2. Allow uploading a new file

Fallback if the ideal case is not possible
 
1. The form page should be reloaded with a message at the top indicating that the CSV was missing data. I can supply a message if we go down this route.

**Additional context**
When we did the registration form redesign, I remember we had a lot of trouble handling what happens if there is an error AFTER the user presses the "Register" button. In most cases with the form fill, we prevent the user from pressing Register if the data to be submitted is invalid, but in the CSV case, we have to parse the CSV before we can tell if it's invalid.

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Why was this change made?

Fixes #4323

- Prefer "docker compose" to docker-compose to be more modern
- Fix bug that wipes out existing form values with invalid CSV uploads

# How was this change tested?

TODO

### CODE REVIEW

File: README.md
Comment (mjgiarlo) [inline README.md:59]: We've been steadily updating documentation to reference the modern incantation for Docker Compose, so I did the same here.

File: README.md
Comment (mjgiarlo) [inline README.md:67]: This is the shortest way to ask Argo's `docker compose` configuration to spin up all services *except* the web container.

File: app/controllers/registrations_controller.rb
Comment (mjgiarlo) [inline app/controllers/registrations_controller.rb:101]: The value of this local var is identical to that of the return value of the `create_params` method, so I suspect this was extracted at some point and we missed this line.

File: app/controllers/registrations_controller.rb
Comment (mjgiarlo) [inline app/controllers/registrations_controller.rb:117]: I'm not sure these were serving a purpose.

File: app/javascript/controllers/registration_item_row_controller.js
Comment (mjgiarlo) [inline app/javascript/controllers/registration_item_row_controller.js:7]: Both the registration item row and registration tabs controllers need to know if the CSV side of the form has been toggled so that the fields on the user-input side of the form won't be validated. (They're blank.)

File: app/javascript/controllers/registration_tabs_controller.js
Comment (mjgiarlo) [inline app/javascript/controllers/registration_tabs_controller.js:17]: Cruft removal.

File: app/javascript/controllers/registration_tabs_controller.js
Comment (mjgiarlo) [inline app/javascript/controllers/registration_tabs_controller.js:13]: See comment.

File: app/views/registrations/show.html.erb
Comment (mjgiarlo) [inline app/views/registrations/show.html.erb:65]: Move this up a few lines so it's available in the `<div>` that mounts the registration-tabs controller.

File: app/controllers/registrations_controller.rb
Comment (mjgiarlo) [inline app/controllers/registrations_controller.rb:126]: Here's where we use the user's entered values to prepopulate the form.

File: app/views/registrations/show.html.erb
Comment (mjgiarlo) [inline app/views/registrations/show.html.erb:4]: See comment. This was another bug hiding in the code.

File: spec/features/item_registration_csv_spec.rb
Comment (mjgiarlo) [inline spec/features/item_registration_csv_spec.rb:116]: Expand scope of test to test what PO expects.

File: spec/features/item_registration_csv_spec.rb
Comment (mjgiarlo) [inline spec/features/item_registration_csv_spec.rb:158]: This makes sure that what the user entered in the form was retained even after an invalid CSV file was uploaded.

**Review by peetucket — APPROVED** [review]
Approved pending test on stage

File: app/javascript/controllers/registration_tabs_controller.js
Comment (ndushay) [inline app/javascript/controllers/registration_tabs_controller.js:13]: rewording suggestion. (I found it hard to grok what you wrote -- too many clauses?):

used only at connect() time when the `csv` value has been set in the HTML (which only occurs after the CSV form has been submitted).  Useful for CSV upload errors.

File: app/views/registrations/show.html.erb
Comment (ndushay) [inline app/views/registrations/show.html.erb:4]: remove "added" -- that's only relevant in the context of this PR;  after merge, it won't be "added" any more -- just part of the code.

File: spec/features/item_registration_csv_spec.rb
Comment (ndushay) [inline spec/features/item_registration_csv_spec.rb:6]: quibble:  I'd have user be the first "let" statement.  No biggie

File: spec/features/item_registration_csv_spec.rb
Comment (ndushay) [inline spec/features/item_registration_csv_spec.rb:118]: is it useful to have a separate test with these values?  Not sure why we needed the change.  Simplifying?

**Review by ndushay — APPROVED** [review]
some quibbles and thoughts in comments.  approving.

File: README.md
Comment (mjgiarlo) [reply README.md:67]: @ndushay https://github.com/sul-dlss/argo/pull/3702 is where that service was added. This change only modifies the README to have a more complete way to spin up services.

File: app/controllers/registrations_controller.rb
Comment (mjgiarlo) [reply app/controllers/registrations_controller.rb:101]: The values are identical 🤷🏻‍♂️

File: spec/features/item_registration_csv_spec.rb
Comment (mjgiarlo) [reply spec/features/item_registration_csv_spec.rb:118]: Choosing more non-default values to make sure they don't default when CSV validation fails.

File: app/javascript/controllers/registration_tabs_controller.js
Comment (mjgiarlo) [reply app/javascript/controllers/registration_tabs_controller.js]: Done

File: app/views/registrations/show.html.erb
Comment (mjgiarlo) [reply app/views/registrations/show.html.erb]: Done

**Pull request merged**
